### ISSUE TITLE
`tree-sitter generate` returns permission error without printing the file name

### ISSUE THREAD

**Issue by atsmtat** [tree-sitter/tree-sitter#2249]
tree-sitter supports reading grammar from both .js and .json files. When `tree-sitter generate` loads grammar from the default `grammar.js` file, it converts it into JSON and then writes it into `src/grammar.json` file (to keep it consistent with the js version I assume). But if the `src/grammar.json` file doesn't have a write permission, it fails with `Permission denied` error without printing the file name, which leaves the user confused about which file operation failed. I had to find the file name by running strace.

With a simple fix, I was able to get it to print the file name. Let me know if you'd like a PR.

**Comment by ObserverOfTime** [comment]
Yes, a PR would be welcome.

**Comment by WillLillis** [comment]
It looks like this was addressed in #2411.

<Details>

<Summary>Repro with tree-sitter-asm</Summary>

```shell
[lillis@LaptopWill] ~/projects/tree-sitter-asm (main)
❯ nvim grammar.js

[lillis@LaptopWill] ~/projects/tree-sitter-asm (main) ⚡
❯ tree-sitter generate

[lillis@LaptopWill] ~/projects/tree-sitter-asm (main) ⚡
❯ git status
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   grammar.js
        modified:   src/grammar.json

no changes added to commit (use "git add" and/or "git commit -a")

[lillis@LaptopWill] ~/projects/tree-sitter-asm (main) ⚡
❯ chmod -w src/grammar.json

[lillis@LaptopWill] ~/projects/tree-sitter-asm (main) ⚡
❯ git restore grammar.js

[lillis@LaptopWill] ~/projects/tree-sitter-asm (main) ⚡
❯ tree-sitter generate
Failed to write grammar.json to "/home/lillis/projects/tree-sitter-asm/src"

Caused by:
    Permission denied (os error 13)
```

</Details>
